embark-anyone.lua

```diff
@@ -0,0 +1,108 @@
+local dialogs = require 'gui.dialogs'
+local choices = {}
+
+function addCivToEmbarkList(info)
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+   viewscreen.start_civ:insert ('#', info.civ)
+   viewscreen.start_civ_nem_num:insert ('#', info.nemeses)
+   viewscreen.start_civ_entpop_num:insert ('#', info.pops)
+   viewscreen.start_civ_site_num:insert ('#', info.sites)
+end
+
+function embarkAnyone()
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+
+   if viewscreen._type ~= df.viewscreen_choose_start_sitest then
+      qerror("This script can only be used on the embark screen!")
+   end
+
+   for i, civ in ipairs (df.global.world.entities.all) do
+      if civ.type == df.historical_entity_type.Civilization then
+         local available = false
+
+         -- Check if civ is already available to embark
+         for _, item in ipairs (viewscreen.start_civ) do
+            if item == civ then
+               available = true
+               break
+            end
+         end
+
+         if not available then
+            local sites = 0
+            local pops = 0
+            local nemeses = 0
+            local histfigs = 0
+            local label = ''
+
+            -- Civs keep links to sites they no longer hold, so check owner
+            -- We also take the opportunity to count population
+            for j, link in ipairs(civ.site_links) do
+               local site = df.global.world.world_data.sites[link.target]
```

Comment:
if `link.target` is the site id, you'd get the site with `df.site.find(link.target)`, which does a binary search through the sites vector. the ID is not necessarily the same as the offset.